.github/workflows/docker.yml

```diff
@@ -0,0 +1,49 @@
+name: Docker Publish
+
+on:
+  push:
+    branches: [ feature/docker-and-package-upgrades ]
+
+jobs:
+  build-and-push-docker-image:
+    # Publish to GitHub Container Registry
+    # Master branch only
+    # if: ${{ github.ref == 'refs/heads/master' }}
+    # needs:
+    #  - tests
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+    env: 
+      REGISTRY: ghcr.io
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@v2
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract metadata (tags, labels) for OpenActive Test Suite
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: ${{ env.REGISTRY }}/openactive/test-suite
+          labels: |
+            org.opencontainers.image.title=OpenActive Test Suite
+            org.opencontainers.image.description=Test suite for OpenActive data publishing and Open Booking API implementations
+            org.opencontainers.image.vendor=OpenActive
+          tags: |
+            latest
+
+      - name: Build and push Docker image for OpenActive Test Suite
+        uses: docker/build-push-action@v4
+        with:
+          context: .
+          push: true
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
```

Comment:
_(Copying Nick's comment from: https://github.com/openactive/openactive-test-suite/pull/553#discussion_r1330595422)_

This needs to be expanded to test before push (https://docs.docker.com/build/ci/github-actions/test-before-push/), to ensure that everything works as expected.

The reference implementation service (https://github.com/openactive/OpenActive.Server.NET/blob/d6e2f57dd8b6a7835cf8bcd747eef7553a7128f4/.github/workflows/docker-test.yaml#L16-L32) can be added here to facilitate this.

The config can also be passed in (https://github.com/openactive/OpenActive.Server.NET/blob/d6e2f57dd8b6a7835cf8bcd747eef7553a7128f4/.github/workflows/docker-test.yaml#L53-L56)

Though note that we'd need to get [ghcr.io/openactive/reference.bookingsystem.identity](http://ghcr.io/openactive/reference.bookingsystem.identity) running to actually test Puppeteer (which requires figuring out how to get it running on HTTP, perhaps by overriding the "urls" property in appsettings.json via an env var as per https://github.com/dotnet/AspNetCore.Docs/issues/25626#issuecomment-1098616664)

Also note that the two use the same fake database and there may be assumptions tied to this running on the same machine (as it writes to the same location on local disk), therefore something like this may be more appropriate https://github.com/openactive/OpenActive.Server.NET/blob/d6e2f57dd8b6a7835cf8bcd747eef7553a7128f4/.github/workflows/docker-test-2.yaml#L18-L38

Reply:
I'll move this to a separate issue. I don't think it's critical because we already have CI running on all PRs that go into `master` and branch protection rules forbidding merging until the CI passes

Reply:
That issue: https://github.com/openactive/openactive-test-suite/issues/681